tests/numpy_test.py

```diff
@@ -106,6 +109,146 @@ def test_ndarray_roundtrip(self):
             assert_equal(decoded, array)
             self.assertEqual(decoded.dtype, array.dtype)
 
+    def test_shape(self):
+        """test that shapes containing zeros, which cannot be represented as nested lists, are deserialized correctly"""
+        a = np.eye(3)[3:]
+        _a = self.roundtrip(a)
+        npt.assert_array_equal(a, _a)
+
+    def test_accuracy(self):
+        """test if the string representation maintains accuracy"""
+        rand = np.random.randn(3, 3)
+        _rand = self.roundtrip(rand)
+        npt.assert_array_equal(rand, _rand)
+
+    def test_b64(self):
+        """test that binary encoding works"""
+        a = np.random.rand(10, 10)    # array of substantial size is stored as b64
+        _a = self.roundtrip(a)
+        print(jsonpickle.encode(a))
+        npt.assert_array_equal(a, _a)
+
+    def test_views(self):
+        """Test that views are maintained under serialization"""
+        rng = np.arange(20)  # a range of an array
+        view = rng[10:]  # a view referencing a portion of an array
+        data = [rng, view]
+
+        _data = self.roundtrip(data)
+
+        _data[0][15] = -1
+        assert _data[1][5] == -1
```

Comment:
Ditto for the rest of the asserts below. The python `self.assertEqual(...)`, `self.assertTrue(...)` family of functions are preferred over `assert` because the latter is optimized away when running python in optimized mode, and generally the suggested way to write test code.